manifests/caching.pp

```diff
@@ -10,8 +10,8 @@
 # @author https://github.com/simp/pupmod-simp-named/graphs/contributors
 #
 class named::caching(
-  Stdlib::Absolutepath $chroot_path = $::named::params::chroot_path
-) inherits ::named::params {
+  Stdlib::Absolutepath $chroot_path = simplib::lookup('simp_options::named::chroot', { 'default_value' => '/var/named/chroot' }),
```

Comment:
@TheSemicolons So, the general use of `simp_options` is to provide globally-useful catalysts to various puppet modules that can (optionally) do SIMP-specific things (think iptables, auditing, syslog, etc...).

This allows us to make the state of the system more user-driven and less mysterious in general.

For something like this, we opted for using `params.pp` since we needed to prevent issues with inheritance between classes that are unable to co-exist since they manage the same resources and are effectively two different system states.

That said, you are correct in that there is probably a better way to do this now.

What I would recommend is the following:
* Remove `params.pp`
* Move the item into [data in modules](https://puppet.com/docs/puppet/5.1/hiera_migrate_modules.html)
* Set `named::chroot_path` to `/var/named/chroot` via the Hiera data in the module
* Set `named::caching::chroot_path` to be an alias to `named::chroot_path`

This means that Hiera will work like usual and there won't be any mystery between the two classes any longer.

As it currently stands, Hiera will still work like usual but you would need to set each setting separately. With the data in modules version, if you sest `named::chroot_path`, it will "just work" even if you switch over to caching mode which is a touch magical but probably what most people would want.

Reply:
So this will add a `pupmod-simp-named/hiera.yaml` and `pupmod-simp-named/data/common.yaml`, the latter of which is where `named::chroot_path` will be set?

Reply:
@TheSemicolons You've got it!

Reply:
@trevor-vaughan Updated. I couldn't figure out how to add an alias for named::caching::chroot without more code changes that I didn't understand. Since both paths must be the same, I just left it as named::chroot.